internal/pkg/service/stream/storage/level/local/diskreader/reader.go

```diff
@@ -67,8 +69,27 @@ func newReader(
 	ctx = ctxattr.ContextWith(ctx, attribute.String("slice", sliceKey.String()))
 	r.logger.Debug(ctx, "opening disk reader")
 
+	matched, err := filepath.Glob(path)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(matched) == 0 {
+		err = errors.New("no matched path to read")
```

Comment:
Something better would be useful here, ... e.g. `no data`/`no slice part file found` ...